### DISCUSSION TITLE
extending brew test timeout

### DISCUSSION THREAD

**Discussion by mahrud** [Homebrew/discussions#5692] [Tap maintenance and Homebrew development] [answered]
Is there a way to change the default test timeout in taps?
https://github.com/Homebrew/brew/blob/f921bb4e345e083f4966c9e64b608b584c2a646a/Library/Homebrew/test.rb#L19

**Comment by gromgit** [comment]
Not that I'm aware of. If you feel that there's a general need to test for longer than 5 minutes, you could [open a PR](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) that implements the necessary, maybe add a `HOMEBREW_TEST_TIMEOUT_SECS` environment variable that overrides that default.

**Reply by mahrud** [reply]
Thanks! I think I'll leave this question here to see if anyone else is also interested in this feature as well.

**Comment by MikeMcQuaid** [comment]
@mahrud can you explain why you'd like longer tests?

**Reply by mahrud** [reply]
It's not that we have very long tests, but we have a _ton_ of tests that that altogether take about an hour and a half! I certainly don't need to run them on every rebuild or even locally with `brew test` (that's what our own [workflows](https://github.com/Macaulay2/M2/actions/workflows/test_build.yml) are for), but every new release it would be nice to run all tests on all bottles just _once_ to make sure everything works.

**Reply by MikeMcQuaid** [reply, marked as answer]
@mahrud If you're not wanting to run on every rebuild or locally: this doesn't sound like the best use of `brew test` for me but instead e.g. an additional script in your release workflows?

**Reply by mahrud** [reply]
Ah, sure, that's reasonable. I can add this to the pr-pull workflow. Thanks!
